subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/result/DefaultResolvedArtifactResult.java

```diff
@@ -33,11 +34,12 @@
 
     public DefaultResolvedArtifactResult(ComponentArtifactIdentifier identifier,
                                          AttributeContainer variantAttributes,
+                                         List<? extends Capability> capabilities,
                                          DisplayName variantDisplayName,
                                          Class<? extends Artifact> type,
                                          File file) {
         this.identifier = identifier;
-        this.variant = new DefaultResolvedVariantResult(identifier.getComponentIdentifier(), variantDisplayName, variantAttributes, Collections.emptyList(), null);
```

Comment:
And this was where we simply ignored the fact that the capability information should have been available and not default to an empty list.